clang/test/SemaSYCL/device_has_ast.cpp

```diff
@@ -71,7 +75,18 @@ class KernelFunctor {
 
 void foo() {
   q.submit([&](handler &h) {
+    // Attribute applied on operator of kernel functor should be applied on kernel function.
+    // CHECK: FunctionDecl {{.*}}kernel_name_1
+    // CHECK: SYCLDeviceHasAttr
+    // CHECK-NEXT: DeclRefExpr {{.*}} 'sycl::aspect' EnumConstant {{.*}} 'cpu' 'sycl::aspect'
+    // Attributes applied to functions called from the kernel should not be propagated to kernel.
+    // CHECK-NOT: SYCLDeviceHasAttr
```

Comment:
I'm not quite following the comment and the `CHECK-NOT` line. The kernel already has the attribute.
Is this `CHECK-NOT` line intended to make sure that called functions don't have the attribute?

Reply:
I also do not follow this one. The kernel has attribute. i agree with @Fznamznon.
@elizabethandrews, I think it would be clear if you add tests like this 
```
// Test attribute is not propagated.
    h.single_task<class kernel_name3>(
        []() { fun1c(); });